### PR TITLE
[[ Widgets ]] Factor out common handlers into utility library

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

**Comment by livecodeali** [comment]
Requires this https://github.com/livecode/livecode-ide/pull/511

File: extensions/modules/widget-utils/widget-utils.lcb [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+Description:
+Use the <stringToColor> handler when receiving colors from LiveCode script 
+via a property setter, or from the widget's stored properties array when loading.
+*/
+public handler stringToColor(in pString as String) returns Color
+	variable tRed as Real
+	variable tGreen as Real
+	variable tBlue as Real
+	variable tAlpha as Real
```
Comment (livecodeian) [inline extensions/modules/widget-utils/widget-utils.lcb]: Unused variables tRed, tGreen, tBlue, tAlpha

File: extensions/modules/widget-utils/widget-utils.lcb [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	variable tColor as List
+	put [] into tColor
+	variable tColorComponent as String
+	repeat for each element tColorComponent in tComponentList
+		push stringToColorComponent(tColorComponent) onto tColor
+	end repeat
+
+	if tComponentCount is 3 then
+		push 1 onto tColor
+	end if
```
Comment (livecodeian) [inline extensions/modules/widget-utils/widget-utils.lcb]: This isn't necessary - color [r, b, g] will set the alpha component to 1 by default

File: extensions/modules/widget-utils/widget-utils.lcb [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+public handler stringToColor(in pString as String) returns Color
+	variable tComponentList as List
+	split pString by "," into tComponentList
+
+	variable tComponentCount
+	put the number of elements in tComponentList into tComponentCount
+	if tComponentCount is not 3 and tComponentCount is not 4 then
+		// Invalid number of components detected
+		throw "Invalid color"
+	end if
```
Comment (livecodeian) [inline extensions/modules/widget-utils/widget-utils.lcb]: This check isn't strictly necessary, as "color tColor" will throw an error if the list doesn't contain 3 or 4 numbers.
Sorry I missed this earlier.

**Comment by livecodeian** [comment]
@livecode-vulcan review ok 65be02b

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok 65be02bc432068f8dd7036a274411704bd7479ec

**Comment by livecode-vulcan** [comment]
😎 test success 65be02bc432068f8dd7036a274411704bd7479ec
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/537)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/541)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/536)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/536)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/545)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/539)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/547)

**Pull request merged**